File: chrome/browser/ui/browser_window/internal/browser_window_interface_iterator_android.cc
// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "chrome/browser/ui/browser_window/public/browser_window_interface_iterator.h"

#include "base/notimplemented.h"

std::vector<BrowserWindowInterface*> GetAllBrowserWindowInterfaces() {
  // TODO(https://crbug.com/419057482): Implement this once we have an
  // Android variant of BrowserWindowInterface.
  NOTIMPLEMENTED();
  return {};
}

std::vector<BrowserWindowInterface*>
GetBrowserWindowInterfacesOrderedByActivation() {
  // TODO(https://crbug.com/419057482): Implement this once we have an
  // Android variant of BrowserWindowInterface.
  NOTIMPLEMENTED();
  return {};
}

File: chrome/browser/ui/browser_window/internal/android/java/src/org/chromium/chrome/browser/ui/browser_window/BrowserWindowInterfaceIteratorAndroid.java
// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.chromium.chrome.browser.ui.browser_window;

import org.jni_zero.CalledByNative;
import org.jni_zero.JniType;

import org.chromium.build.annotations.NullMarked;

/**
 * Java class implementing the native functions in {@code
 * browser_window_interface_iterator_android.cc}.
 */
@NullMarked
final class BrowserWindowInterfaceIteratorAndroid {
    private BrowserWindowInterfaceIteratorAndroid() {}

    @CalledByNative
    @JniType("std::vector<int64_t>")
    private static long[] getAllBrowserWindowInterfaces() {
        return ChromeAndroidTaskTrackerImpl.getInstance().getAllNativeBrowserWindowPtrs();
    }

    @CalledByNative
    @JniType("std::vector<int64_t>")
    private static long[] getBrowserWindowInterfacesOrderedByActivation() {
        return ChromeAndroidTaskTrackerImpl.getInstance()
                .getNativeBrowserWindowPtrsOrderedByActivation();
    }
}
